### ISSUE TITLE
Invalid "overloaded function signature will never be matched"

### ISSUE THREAD

**Issue by Holt59** [python/mypy#15998]
**Bug Report**

When trying to use `typing.overload` with frames from the astropy project (https://www.astropy.org/), incorrect errors are reported by mypy.

**To Reproduce**

```python
from typing import overload

from astropy.coordinates import GCRS, ITRS


@overload
def f(frame: ITRS) -> None:
    ...


@overload
def f(frame: GCRS) -> None:
    ...


def f(frame: ITRS | GCRS) -> None:
    ...

```

**Expected Behavior**

No error.

**Actual Behavior**

Second overload generates

```
test.py:12: error: Overloaded function signature 2 will never be matched: signature 1's parameter type(s) are the same or broader  [misc]
```

**Your Environment**

- Mypy version used: 1.151
- Mypy command-line flags: 
- Mypy configuration options from `mypy.ini` (and other config files): `warn_return_any = true` and `warn_unused_configs = true`
- Python version used: 3.10.5

**Comment by JelleZijlstra** [comment]
Is astropy an untyped library without stubs? If so, both of the types are `Any`, so from mypy's perspective the two overloads are indeed equivalent.

**Comment by Holt59** [comment]
> Is astropy an untyped library without stubs? If so, both of the types are `Any`, so from mypy's perspective the two overloads are indeed equivalent.

Yes `astropy` does not have stubs. Is there anyway to ignore this particular error on this line (without completely disabling typing for the line)?
